container-storage-setup.sh

```diff
@@ -731,40 +729,47 @@ create_partition_parted(){
 }
 
 create_partition() {
-  local dev="$1"
+  local dev="$1" part
 
   if [ -x "/usr/sbin/parted" ]; then
     create_partition_parted $dev
   else
     create_partition_sfdisk $dev
   fi
 
+  part=$(query_first_child $dev)
```

Comment:
If this succeeds, then the partition exists and there is no point in calling `udevadm settle` or `wait_for_dev` below.  If you are worried that you need to wait for the partition block device to come into existence after calling `sfdisk`, then you should probably call `query_first_child` in a loop until it returns something non-empty.

However, the device for the partition does exist after calling `sfdisk` so there is no need to wait for it to appear.  What does happen, though, is that udevd asynchronously calls the BLKRRPART ioctl some time after the partition has been created, and because of the way BLKRRPART works, there is a brief period of time where the block device for the partition disappears.  If you want to protect against that, I recommend wrapping the whole `sfdisk; pvcreate; vgextend` operation in `flock -s $dev`.  The shared lock signals to udevd not to do the BLKRRPART ioctl.